### DISCUSSION TITLE
Compost » fertilizer?

### DISCUSSION THREAD

**Discussion by jmmullins** [jeremyckahn/farmhand#435] [Q&A] [answered]
I have the ability to turn weeds into compost via the recycling tab, and the compost description says it can be used to make fertilizer. I’m apparently not far enough into the game to do that though, so I'm not sure if should be selling the compost at this point or saving it; knowing how far I am from being able to do this would help inform my decision on what to do in the meantime. 

Maybe that description line for compost on the recycling tab could include info about how you can unlock the ability to turn compost into fertilizer and/or what requirements need to be fulfilled to gain that ability please.

**Comment by jeremyckahn** [comment, marked as answer]
You can turn Compost into Fertilizer once you sell 10 Compost units: https://github.com/jeremyckahn/farmhand/blob/d4b00419078efd8fb6f17e2208bd511abf07a795/src/data/recipes.js#L645-L658

(Note lines 651 and 652 in particular.)

I realize now that this isn't super obvious. Good call out! Would it help if we tweak the "Can be used to make fertilizer" text to something like "Can be used to make fertilizer once you sell a lot of it?"

**Comment by jmmullins** [comment]
Yeah, even if we keep it vague by not telling them how many units they need to sell, SOME kind of direction within the compost item description would be excellent. I hunted around all the pages and UI elements a few times trying to work it out. 

Alternatively and/or additionally, since it's tied to doing something specific maybe this should be included in the Achievements list? Something like: 

> Recycling Novice
> Reward: Unlocks ability/recipe to turn compost into fertilizer
> 
> Sell X units of compost [OR Sell a lot of compost (if intentional is to keep it vague)]

**Reply by lstebner** [reply]
Selling X amount of things to unlock something else is a pretty core concept to the game, and I definitely think it could be explained better. I know this question was specific to fertilizer, but basically all recipes work this way and it took me a long time to get used to since I have this natural tendency in games like this to hoard items in case I need them later. I think it could be nice to build some sort of UI that does show there are things to unlock the player is progressing towards.

**Reply by jeremyckahn** [reply]
I think @lstebner is speaking to a broader design shortcoming of the game that's causing the confusion that this thread is about. It's been a longstanding design choice that Farmhand recipes would get unlocked by selling a large enough quantity of their constituent ingredients. This was left intentionally vague to players because I had hoped that unlocking recipes would come as a fun surprise as players sell off various items. However, this falls apart here:

> I have this natural tendency in games like this to hoard items in case I need them later.

I'm the exact same way, which is why I have an absurd amount of potions at the end of any RPG I play. 😅

It's good to have your collective perspectives on this, because I can see now that hiding recipes behind invisible gates doesn't make for a great experience. Tangentially, the surprise of unlocking a recipe gets lost when the player doesn't even notice the notification, which is a separate design issue that Farmhand has.

Considering all of this, I think it would be worth revisiting the UX around unlocking recipes. One idea that comes to mind is showing within individual item cards what recipes they can unlock. We could also have a centralized list of all recipes that show the requirements that are needed to unlock them. Either way, we'd need to rework how recipes are implemented.

---------------

In terms of the technical changes that would need to happen: We'd need to change the structure of recipes internally to define a map of ingredients to quantities, rather than than an arbitrary `condition` function. For example, here's the Salsa recipe:

```js
export const salsa = itemify({
  id: 'salsa',
  name: 'Salsa',
  ingredients: {
    [items.jalapeno.id]: 1,
    [items.onion.id]: 1,
    [items.tomato.id]: 1,
    [items.corn.id]: 1,
  },
  condition: state =>
    state.itemsSold[items.jalapeno.id] >= 5 &&
    state.itemsSold[items.onion.id] >= 5 &&
    state.itemsSold[items.tomato.id] >= 5 &&
    state.itemsSold[items.corn.id] >= 5,
  recipeType: recipeType.KITCHEN,
})
```

I'm thinking we'd need a more parseable `condition`, like this:

```js
export const salsa = itemify({
  id: 'salsa',
  name: 'Salsa',
  ingredients: {
    [items.jalapeno.id]: 1,
    [items.onion.id]: 1,
    [items.tomato.id]: 1,
    [items.corn.id]: 1,
  },
  condition: {
    [items.jalapeno.id]: 5,
    [items.onion.id]: 5,
    [items.tomato.id]: 5,
    [items.corn.id]: 5,
  },
  recipeType: recipeType.KITCHEN,
})
```

Implementing the logic to support this new format would probably be a medium-sized challenge, but updating all of the existing recipes for the new format would be a fairly tedious process (but worth it!). There would also be some edge cases that we'd need to account for, such as `fertilizer`'s `condition`:

```js
condition: state =>
  state.purchasedComposter && state.itemsSold[compost.id] >= 10
```

But I'm sure we could figure out something for this. #439 has been filed to get this done! 🙂
